server/types/app.go

```diff
@@ -0,0 +1,48 @@
+package types
```

Comment:
side note: i moved these types to a separate types package to avoid cyclic imports